Wikipedia/Code/WebViewController.m

```diff
@@ -514,6 +518,41 @@ - (void)moveFindInPageSelectedMatchIndexInDirection:(WMFFindInPageScrollDirectio
     }
 }
 
+- (void)scrollToOffset:(CGPoint)offset animated:(BOOL)animated completion:(dispatch_block_t)completion {
+    assert([NSThread isMainThread]);
+    UIScrollView *scrollView = self.webView.scrollView;
+    dispatch_block_t done = ^{
+        if (completion) {
+            completion();
+        }
+    };
+    if (isnan(offset.x) || isinf(offset.x) || isnan(offset.y) || isinf(offset.y)) {
+        done();
+        return;
+    }
+    if (WMFDistanceBetweenPoints(offset, scrollView.contentOffset) < 2) {
+        done();
+        return;
+    }
+    if (!animated) {
+        [scrollView setContentOffset:offset animated:NO];
+        done();
+        return;
+    }
+    /*
+    Setting scrollView.contentOffset inside of an animation block
+    results in a broken animation https://phabricator.wikimedia.org/T232689
+    Calling [scrollView setContentOffset:offset animated:YES] inside
+    of an animation block fixes the animation but doesn't guarantee
+    the content offset will be updated when the animation's completion
+    block is called.
+    It appears the only reliable way to get a callback after the default
+    animation is to use scrollViewDidEndScrollingAnimation
+    */
+    [self.scrollViewAnimationCompletions addObject:completion];
```

Comment:
I think we should pass done here instead - it'll crash if completion is nil.

Reply:
good catch! fixed